core/templates/dev/head/galleries/Gallery.js

```diff
@@ -203,6 +203,125 @@ oppia.controller('Gallery', [
       $rootScope.loadingMessage = '';
     });
 
+    // Below is the width of each tile
+    var TILEDISPLAYWIDTH = 211.953;
+    // If the value below is changed, the corresponding CSS values must be
+    // changed in oppia.css.
+    var MAXTILEDISPLAY = 4;
+    var maxGalleryTileWidth = TILEDISPLAYWIDTH * MAXTILEDISPLAY;
+    $scope.tileDisplayCount = 0;
+    $scope.galleryTilesResizedWidth = 0;
+
+    var calculateCarouselCssWidth = function() {
```

Comment:
What about calling this initCarousels(), and calling this function once at the outset (when this controller is first loaded), and subsequently in the resize() handler?